state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -243,27 +247,59 @@ pub struct CcmSwapAmounts {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
-enum CcmSwapState {
-	PrincipalSwapScheduled {
-		principal_swap_id: SwapId,
-		gas_budget: AssetAmount,
-		other_gas_asset: Option<Asset>,
-	},
-	GasSwapScheduled {
-		gas_swap_id: SwapId,
-		principal_output_amount: AssetAmount,
-	},
+enum GasSwapState {
+	OutputReady { gas_budget: AssetAmount },
+	Scheduled { gas_swap_id: SwapId },
+	ToBeScheduled { gas_budget: AssetAmount, other_gas_asset: Asset },
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+struct DcaState {
+	scheduled_chunk_swap_id: Option<SwapId>,
+	remaining_input_amount: AssetAmount,
+	remaining_chunks: u32,
+	chunk_interval: u32,
+	accumulated_output_amount: AssetAmount,
+}
+
+impl DcaState {
+	// Create initial DCA state; if no dca parameters provided (for non-DCA swaps),
+	// this creates state equivalent to 1 chunk DCA
+	fn new(input_amount: AssetAmount, params: Option<DcaParameters>) -> DcaState {
```

Comment:
It is used in 2 out of 3 places where DcaState is created. The third one is a special case where we want 0 dca chunks: in case of CCM where only gas swap (but not the principal) is required. I decided constructing the state by hand was justified for that case.